ecosystem/typescript/sdk/src/aptos_types/type_tag.ts

```diff
@@ -218,3 +217,192 @@ export class StructTag {
     return new StructTag(address, moduleName, name, typeArgs);
   }
 }
+
+function bail(message: string) {
+  throw new Error(message);
```

Comment:
Let's avoid generic errors and come up with TypeTagParserError or something?

Reply:
Yep sounds good. I can create a new `TypeTagParserError` which extends the generic Error class